react-components/src/apps/listing/appSorting.js

```diff
@@ -0,0 +1,32 @@
+import AppFields from "./AppFields";
+
+const desc = (a, b, orderBy) => {
+    if (
+        orderBy === AppFields.NAME.key ||
+        orderBy === AppFields.INTEGRATOR.key ||
+        orderBy === AppFields.SYSTEM.key
+    ) {
+        if (b[`${orderBy}`] < a[`${orderBy}`]) {
+            return -1;
+        }
+        if (b[`${orderBy}`] > a[`${orderBy}`]) {
+            return 1;
+        }
+        return 0;
+    }
+    if (orderBy === AppFields.RATING.key) {
+        if (b.rating.average < a.rating.average) {
+            return -1;
+        }
+        if (b.rating.average > a.rating.average) {
+            return 1;
+        }
+        return 0;
+    }
+};
```

Comment:
This is copy of the generic sorting algo from ui-lib. But I have added logic to sort rating (nested key).

Reply:
😖 I didn't realize it doesn't work for nested keys.  Maybe we can look into updating it in `ui-lib` to work properly in a different PR.

Reply:
Good point. I can just call the service with sorting option. Thanks for reminding about that.